components/gif_picker/components/Categories/index.jsx

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+
+import {requestCategoriesList, requestCategoriesListIfNeeded, saveSearchBarText, saveSearchScrollPosition, searchTextUpdate, sendViewCountBatch} from 'mattermost-redux/actions/gifs';
+import {changeOpacity, makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import store from 'stores/redux_store.jsx';
+import Constants from 'utils/constants';
+
+import InfiniteScroll from 'components/gif_picker/components/InfiniteScroll';
+
+import './Categories.scss';
+
+function mapStateToProps(state) {
+    return {
+        ...state.entities.gifs.categories,
+        ...state.entities.gifs.cache,
+        appProps: state.entities.gifs.app,
+        searchText: state.entities.gifs.search.searchText,
+        searchBarText: state.entities.gifs.search.searchBarText,
+    };
+}
+
+const mapDispatchToProps = ({
+    saveSearchBarText,
+    saveSearchScrollPosition,
+    searchTextUpdate,
+    requestCategoriesList,
+    requestCategoriesListIfNeeded,
+    sendViewCountBatch,
+});
+
+const getStyle = makeStyleFromTheme((theme) => {
+    return {
+        background: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.05),
+        },
+    };
+});
+
+export class Categories extends Component {
+    static propTypes = {
+        appProps: PropTypes.object,
+        gifs: PropTypes.object,
+        hasMore: PropTypes.bool,
+        onSearch: PropTypes.func,
+        onTrending: PropTypes.func,
+        requestCategoriesList: PropTypes.func,
+        requestCategoriesListIfNeeded: PropTypes.func,
+        saveSearchBarText: PropTypes.func,
+        saveSearchScrollPosition: PropTypes.func,
+        searchTextUpdate: PropTypes.func,
+        sendViewCountBatch: PropTypes.func,
+        searchBarText: PropTypes.string,
+        tagsList: PropTypes.array,
+    }
+
+    componentDidMount() {
+        window.scrollTo(0, 0);
```

Comment:
What exactly does this do? I'm imaging the goal is to scroll just the component in question, but this seems like it scrolls the whole document? Or this is a fullscreen component?